### PR TITLE
Breaking changes for type removal

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
This PR contains the first set of breaking changes for 2.0 related to OpenSearch 2.0 - https://github.com/opensearch-project/OpenSearch/issues/2480

The following changes are included in this PR:

1. A CI workflow `test-integration-unreleased.yml` which runs integ tests for the client against unreleased OpenSearch. This will help in future development to prepare clients with the latest unreleased OpenSearch versions. Currently this workflow will test against 2.0 OpenSearch.
2. The CI workflow `test-integration.yml` is updated to work for released versions which the client is compatible with.
3. Upgraded the OpenSearch version in the client to 2.0.0-alpha1-SNAPSHOT.
4. Type mapping removal across the APIs and related test changes.
5. Remove `indices.exists_type` and `indices.flush_synced` and related files.
 
### Issues Resolved
Part of #139 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/build.yml
Comment (reta) [inline .github/workflows/build.yml:21]: Hm ... why we are building `opensearch-project/opensearch` instead of using published snapshots from `https://aws.oss.sonatype.org/content/repositories/snapshots/`?

File: java-client/src/main/java/org/opensearch/client/opensearch/core/BulkRequest.java
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/opensearch/core/BulkRequest.java:586]: Seems like `_index` should not be dropped

Comment (VachaShah) [reply java-client/src/main/java/org/opensearch/client/opensearch/core/BulkRequest.java:586]: The `_index` part is before this if condition. https://github.com/opensearch-project/opensearch-java/pull/132/files/1e01fbeae03463bf2d47b8d395bce3701901448d#diff-2bd95eccdb4274c5118be928fb9f466034f19fc2cdcfc3e5c82609585af94e0aL576

Comment (reta) [reply java-client/src/main/java/org/opensearch/client/opensearch/core/BulkRequest.java:586]: Ah ... not visible in changeset, thank you!

File: .github/workflows/build.yml
Comment (VachaShah) [reply .github/workflows/build.yml:21]: Ah my bad! I can use that for build and unit test. I will update the workflows for the same.

File: .github/workflows/test-integration-unreleased.yml
Comment (reta) [inline .github/workflows/test-integration-unreleased.yml:43]: I am 100% sure there is a staging repository for Docker images, @dblock @peterzhuamazon could you please refer to the right one?

Comment (VachaShah) [reply .github/workflows/test-integration-unreleased.yml:43]: Yes there is one, but the staging image is with all the plugins which isn't ready yet. This is to test against min OpenSearch image.

Comment (VachaShah) [reply .github/workflows/test-integration-unreleased.yml:43]: I also tried the route of writing Dockerfile and related things to create a docker image for min OpenSearch. But then I found this gradle task that runs in assemble for OpenSearch so used this one. LMK if you think the Dockerfile route is better. I can use that instead.

**Comment by VachaShah** [comment]
I will merge the changes from #134 once that is approved and merged and update the workflows with the appropriate JDK versions.

File: .github/workflows/test-integration-unreleased.yml
Comment (reta) [reply .github/workflows/test-integration-unreleased.yml:43]: I think publishing min Docker images (along with artifacts) would be the best option, may be we could quickly address that? Another possibility is to use OpenSearch test scaffolding (available as set of Gradle plugins) to run clusters from the test suites. What do you think?

Comment (VachaShah) [reply .github/workflows/test-integration-unreleased.yml:43]: I didn't use the Gradle plugins since I wanted to create a CI way for all the clients. It can work for java client but we would still end up in the same place for other clients.

Comment (reta) [reply .github/workflows/test-integration-unreleased.yml:43]: Agree, thank you, publishing min Docker images (along with artifacts) would be the answer, right?

Comment (VachaShah) [reply .github/workflows/test-integration-unreleased.yml:43]: Yes that would be the best option. Till that happens, is it fine if we use this change so we can start the development for 2.0? WDYT?

**Review by reta — APPROVED** [review]

**Review by CEHENKLE — APPROVED** [review]

File: .github/workflows/test-integration-unreleased.yml
Comment (VijayanB) [reply .github/workflows/test-integration-unreleased.yml:43]: I believe we will be adding API for plugins too and we need unreleased complete distribution for testing. Shall we create an issue to track this?

Comment (reta) [reply .github/workflows/test-integration-unreleased.yml:43]: Yes, please do, I think it will useful for many

**Comment by VachaShah** [comment]
@reta @CEHENKLE @VijayanB A review again please? Just merged the changes from the main branch for the workflows and removed `indices.exists_type` as part of the breaking changes.

**Review by CEHENKLE — APPROVED** [review]

**Review by reta — APPROVED** [review]

**Comment by owaiskazi19** [comment]
@VachaShah I see java client is using checkstyle for formatting. Moving forward do we have any plan to have Spotless as the formatting tool?

**Comment by VachaShah** [comment]
> @VachaShah I see java client is using checkstyle for formatting. Moving forward do we have any plan to have Spotless as the formatting tool?

No plans as such but you can create a PR :)

**Review by VijayanB — APPROVED** [review]
LGTM. Thanks for making the changes.

**Comment by VachaShah** [comment]
Not sure why the whitesource check is failing, the dependency was updated in #129. Will keep an eye on it.

File: java-client/build.gradle.kts
Comment (dermot-hardy) [inline java-client/build.gradle.kts:133]: Is this introducing a hard-coded dependency on the Log4j framework?  Surely the client should only be dependent on the facade?

Comment (dermot-hardy) [reply java-client/build.gradle.kts:133]: #175

**Pull request merged**
